File: interpreter/java/DeveloperExpression.java
package Interpreter;

// Dil bilgisindeki terminal ifadeler üzerinde işlem yapar.
// UML diyagramındaki TerminalExpression sınıfına denk gelmektedir.
class DeveloperExpression implements IJobExpression {
    public void interpret(Context context) {
        if (context.formula.contains("G")) {
            context.totalPoint += 4000;
        }
    }
}

File: interpreter/kotlin/DeveloperExpression.kt
package Interpreter

// Dil bilgisindeki terminal ifadeler üzerinde işlem yapar.
// UML diyagramındaki TerminalExpression sınıfına denk gelmektedir.
internal class DeveloperExpression : IJobExpression {
    override fun interpret(context: Context) {
        if (context.formula!!.contains("G")) {
            context.totalPoint = context.totalPoint + 4000
        }
    }
}